### DISCUSSION TITLE
signInWithCredential not throwing an error.

### DISCUSSION THREAD

**Discussion by selven** [invertase/react-native-firebase#4579] [Q&A]
Hey, I have an issue I hope someone can help me with. If a user signs up with an email/ password, signs out, then tries to sign in using google. It doesn't throw the `auth/account-exists-with-different-credential` error I would expect. Instead it overwrites the user in firebase and the user can only ever login with google from now on. Here are the key calls I'm using currently.

**Sign up**
``` js
auth().createUserWithEmailAndPassword(email, password)```
```

**Login**
``` js
auth().signInWithEmailAndPassword(email, password)
```

**Login with google:**
``` js
    const { idToken, user } = await GoogleSignin.signIn().catch((errorMsg) => {
      console.log('error', errorMsg);
    });

    // my current workaround
    const existing = await auth().fetchSignInMethodsForEmail(user.email);
    if (existing.includes('password')) {
      setError(
        'You have already created an account using a username and password',
      );
      return
    }

    // Create a Google credential with the token
    const googleCredential = auth.GoogleAuthProvider.credential(idToken);

    return auth()
      .signInWithCredential(googleCredential)
      .catch((errorMsg) => {
        console.log('error', errorMsg);
      });
```

Am I doing something incorrectly or should I raise this as a bug?

**Comment by mikehardy** [comment]
I'm not sure - I do the same thing where I always fetch sign in methods first, but it is very surprising that despite setting a password they can no longer use it even if the email is the same? This sounds like it would be an upstream issue and an attempt to reproduce (and log if reproduced) using an upstream quickstart for clean repro might help: https://github.com/firebase/quickstart-ios/tree/master/authentication or https://github.com/firebase/quickstart-android/tree/master/auth depending on your preference

**Comment by selven** [comment]
I spent some more time on this today. This only seems to happen if the user has an unverified account.
If the user verifies their email the account wont be replaced and will be linked like I expected.
